go-controller/pkg/ovn/namespace.go

```diff
@@ -410,14 +412,17 @@ func (oc *DefaultNetworkController) deleteNamespace(ns *kapi.Namespace) error {
 
 // getNamespaceLocked locks namespacesMutex, looks up ns, and (if found), returns it with
 // its mutex locked. If ns is not known, nil will be returned
-func (oc *DefaultNetworkController) getNamespaceLocked(ns string, readOnly bool) (*namespaceInfo, func()) {
+func (nsm *namespaceManager) getNamespaceLocked(ns string, readOnly bool) (*namespaceInfo, func()) {
+	if nsm == nil {
+		return nil, nil
```

Comment:
won't this complicate the caller ? 

Could you elaborate into how this scenario can be reached ? Wondering if we shouldn't return an error in this situation.

Reply:
this is only because network policy is not supported for secondary networks yet. once that support is added, every network needs to handle watchNamespace() and will nsm won't be nil. 

but at this time, this should success for networks that do not handle namespace.

Reply:
then I would flat out remove it. 

Thanks for explaining.